tests/integration-tests/tests/cfn-init/test_cfn_init.py

```diff
@@ -54,6 +54,28 @@ def test_replace_compute_on_failure(
     assert_instance_replaced_or_terminating(instance_id, region)
 
 
+@pytest.mark.dimensions("us-west-1", "c5.xlarge", "centos7", "slurm")
+@pytest.mark.usefixtures("os", "instance", "scheduler")
+def test_install_args_quotes(region, pcluster_config_reader, clusters_factory, s3_bucket_factory, test_datadir):
+    """
+    Test pre/post install args with single quote and double quotes.
+
+    The cluster should be created and running.
+    """
+    # Create S3 bucket for pre/post install scripts
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / "pre_install.sh"), "scripts/pre_install.sh")
+    bucket.upload_file(str(test_datadir / "post_install.sh"), "scripts/post_install.sh")
+
+    # Create cluster with initial configuration
+    init_config_file = pcluster_config_reader(bucket_name=bucket_name)
+    cluster = clusters_factory(init_config_file)
+
+    # Check master and compute node status
+    _assert_server_status(cluster)
```

Comment:
Should we assert on the actual args we see to make sure the args are `R curl wget` and not `"R curl wget"`.
Your pre/post install will not trigger an error even if `"R curl wget"` is passed into scripts as a single argument

Reply:
good point, done

Reply:
Let's do some test runs locally to make sure this test fails for existing release version, i.e. it can catch the failure, and passes for your patch